### ISSUE TITLE
Interactive rebase [feature request]?

### ISSUE THREAD

**Issue by i-tengfei** [libgit2/libgit2#6332]
It looks like interactive rebase is not supported.

https://github.com/libgit2/libgit2/blob/main/src/libgit2/rebase.c#L373-L377

https://github.com/libgit2/libgit2/issues/6127

https://github.com/libgit2/libgit2/pull/2482

I want to provide a UI in my program to modify rebase operations to support interactive rebase, but see that libgit2 does not support interactive rebase state.

Will it be supported in the future?

**Comment by ethomson** [comment]
Yes, but in my opinion SHA256 support and fixing the perf for delta resolution are higher priority.

**Comment by akkadaya** [comment]
Any updates on this? apparently `gitui` doesn't support interactive rebase because it's not implemented in this library 🤷‍♂️

**Comment by IEDAWHI** [comment]
Interactive Rebase support would be a nice addition to this library. I also have a use case to build a UI to perform interactive rebases in an application I'm working on.
